snp_application/include/snp_application/bt/snp_bt_ros_nodes.h

```diff
@@ -328,6 +351,19 @@ class CombineTrajectoriesNode : public BT::ControlNode
   BT::NodeStatus tick() override;
 };
 
+class GetCurrentJointStateNode : public BT::RosTopicSubNode<sensor_msgs::msg::JointState>
+{
+public:
+  inline static std::string JOINT_STATE_OUTPUT_PORT_KEY = "current_state";
+  inline static BT::PortsList providedPorts()
+  {
+    return providedBasicPorts({ BT::OutputPort<sensor_msgs::msg::JointState>(JOINT_STATE_OUTPUT_PORT_KEY) });
+  }
+  using BT::RosTopicSubNode<sensor_msgs::msg::JointState>::RosTopicSubNode;
+
+  BT::NodeStatus onTick(const typename sensor_msgs::msg::JointState::SharedPtr& last_msg) override;
+};
```

Comment:
The `ReplaceTrajectoryStartState` also creates a subscriber to the `/joint_states` topic to get the latest joint state. We should consider revising that node to take a joint state as input so we don't have to duplicate this effort and make two subscribers to the same topic